### DISCUSSION TITLE
[Idea] Dissociate Category cover and thumbnail images

### DISCUSSION THREAD

**Discussion by MarsDevPL** [PrestaShop/PrestaShop#32252] [Ideas]
**Describe the bug**
During Category thumbnail editing the bug appears. I am editing Categories images and cat. thumbnails. 
My Theme's Category cover image should have 1003 * 200px. When I chose the  image. it is automatically set as Category thumbnail, and CANNOT cahnge it. In my backoffice i chose Category thumbnail (97 * 128px) and I can only see it in my backoffice, but there is unproperly thumbnail in the front. Whereas when i chose Category thumbnail image in backoffice FIRST, it automatically bacomes the Category cover image. (See screenshots). I deleted cache in backoffice, in browser and on FTP and the problem is still the same. My theme's support respond that it is PRESTASHOP'S 1.7.x.x bug. How to solve this problem?

BackOffice preview
![11](https://user-images.githubusercontent.com/47884714/53631734-854d6d80-3c13-11e9-9cf3-d0a6a3d55750.PNG)
Categoty preview
![22](https://user-images.githubusercontent.com/47884714/53631737-854d6d80-3c13-11e9-8570-58f8761b7ed6.PNG)
subcategory preview
![33](https://user-images.githubusercontent.com/47884714/53631739-854d6d80-3c13-11e9-8ff0-2a736aee09b0.PNG)

**Additionnal information**
PrestaShop version: 1.7.5.0
PHP version: 7.1.21

**Comment by khouloudbelguith** [comment]
Hi @MarsDevPL,

Could you please confirm these steps to reproduce your issue:
1. If I upload a cover image after the category thumbnail, it erases the thumbnail.
2. If I delete the "Category Cover Image" the "Category thumbnail" is deleted too.
3. I tried to add a thumbnail image, in the FO, it is not displayed

In the BO
![image](https://user-images.githubusercontent.com/16067358/53633637-55ed2f80-3c18-11e9-8379-114b50b628b5.png)

In the FO:
![image](https://user-images.githubusercontent.com/16067358/53633668-6d2c1d00-3c18-11e9-898f-19b1e6853ac6.png)

Thanks!

**Comment by MarsDevPL** [comment]
In my case:
1. If I upload a cover image after the category thumbnail;
- In the front I see the cover image instead the thumbnail, but when entering a subcategory - the cover image is showed correctly, but thumbnail is not showing correctly. On the thumbnail's place i see category image with disproportional size.
2. If I delete the "Category Cover Image" the "Category thumbnail" is deleted too. - EXACTLY THE SAME
3. I tried to add a thumbnail image ONLY , in the FO, it is not displayed

**Comment by khouloudbelguith** [comment]
@MarsDevPL, thanks for your feedback.

For the conclusion, I tried with PS1.7.5.1 & PS1.7.4.4

* If I upload a cover image after the category thumbnail, it erases the thumbnail.
* If I upload a thumbnail image after the cover image, the thumbnail image, in the FO is not displayed.
* If I delete the "Category Cover Image" the "Category thumbnail" is deleted too.
* I tried to add a thumbnail image, in the FO, it is not displayed

I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by MarsDevPL** [comment]
I am not so deep into coding, so I just will wait for some solution:)

**Comment by esterflore1** [comment]
Hello, 
is there a solution for this problem?
Thanks a lot for your help

**Comment by MarsDevPL** [comment]
I use the WAREHOUSE theme and i just built the category page with internal
PageBuilder, so I solved this problem in such way.

sob., 30 mar 2019 o 19:16 esterflore1 <notifications@github.com> napisał(a):

> Hello,
> is there a solution for this problem?
> Thanks a lot for your help
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PrestaShop/PrestaShop/issues/12760#issuecomment-478273643>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AtqpqrzEVQ2iPYvzKxGuG0C1Gs4GulL5ks5vb6oVgaJpZM4bYshO>
> .
>


-- 

MarsDevelopment Sp. z o.o
ul. Brochowska 21/10
52-019 Wrocław
NIP 8992810774
REGON 366648341
KRS 0000665452

**Comment by marionf** [comment]
@khouloudbelguith 

> If I upload a cover image after the category thumbnail, it erases the thumbnail.

Yes, because the thumbnail is the same as the category cover image but in smaller

> If I upload a thumbnail image after the cover image, the thumbnail image, in the FO is not displayed.*

Yes, because the thumbnail is not used in classic template

> If I delete the "Category Cover Image" the "Category thumbnail" is deleted too.

Yes, because the thumbnail is the same as the category cover image

These behaviors are normal

**Comment by zenerry** [comment]
> 
> 
> @khouloudbelguith
> 
> > If I upload a cover image after the category thumbnail, it erases the thumbnail.
> 
> Yes, because the thumbnail is the same as the category cover image but in smaller
> 
> > If I upload a thumbnail image after the cover image, the thumbnail image, in the FO is not displayed.*
> 
> Yes, because the thumbnail is not used in classic template
> 
> > If I delete the "Category Cover Image" the "Category thumbnail" is deleted too.
> 
> Yes, because the thumbnail is the same as the category cover image
> 
> These behaviors are normal

Hello folks,
I'd like to comment this thread despite it is closed now,
Is there any way to use "category image" an "thumbnail category image" separetly?
I unsderstand that classic theme doesn't, but other people might try to use them like so (as I'm trying).

Thanks!

**Comment by colinegin** [comment]
Hello all,

after reviewing this issue, it seems there is a bug indeed with how work thumbnail images with cover images.

As said @khouloudbelguith 

- If I upload a cover image after the category thumbnail, it erases the thumbnail and add a new thumbnail with the cover image. (it works well when uploading a thumbnail after the cover image).
- If I delete the "Category Cover Image" the "Category thumbnail" is deleted too.
- It is not possible to delete the thumbnail alone. There is no reason for that.

About those 2 issues @khouloudbelguith , did you test it with Classic theme ? 
- If I upload a thumbnail image after the cover image, the thumbnail image, in the FO is not displayed.
- I tried to add a thumbnail image, in the FO, it is not displayed
==> if so, this is normal, Classic theme in 1.7 does not use category thumbnails.

@marionf, if we offer category thumbnail feature, we should enable merchants to choose to have different cover image and thumbnail. I reopen this issue, and we'll discuss about it to decide what to do !

Thanks everyone for your feedback.

Coline

**Comment by khouloudbelguith** [comment]
Hi @colinegin,

>About those 2 issues @khouloudbelguith , did you test it with Classic theme ?
1.If I upload a thumbnail image after the cover image, the thumbnail image, in the FO is not displayed.
2.I tried to add a thumbnail image, in the FO, it is not displayed
==> if so, this is normal, Classic theme in 1.7 does not use category thumbnails.

Yes, I used the classic theme provided by PrestaShop Version 1.0.0
https://drive.google.com/file/d/120yFPfXsab7RwpvpS8jG7ixr2nB93GWX/view

Thanks!

**Comment by marionf** [comment]
We should re-think the behavior in BO category page

**Comment by HartLarsson** [comment]
[![Screenshot from Gyazo](https://gyazo.com/d986c4795e3d946650322f74bcc7e5a4/raw)](https://gyazo.com/d986c4795e3d946650322f74bcc7e5a4)

Why not add a [Delete] Button here too??

**Comment by generare** [comment]
This seems to be a longlasting request. Is there any extra module that would fix this issue since this is quite challenging for the category image usage. It would be great to have wider category images on the page of the category and more squared-shaped image as a thumbnail.

**Comment by yannicka** [comment]
The problem still exists on PrestaShop 1.7.8.7. Is there a fix?

**Comment by Nalem14** [comment]
This is so ridiculous that this problem is always here on Prestashop 8, since 2019. Else don't allow to edit the thumbnail separatly...

So surprising when you have already set all thumbnails for +100 categories, and you wan't to add cover after... So we need to re set all images now. Absolutely not friendly usage haha

**Comment by JavierEB** [comment]
This was completely forgotten... It still doesn't work in version 8.2.0 and 1.7.8.11 

This is very sad...it should not be abandoned

**Comment by Hlavtox** [comment]
This problem is fixed in **Prestashop 9.0.0** that is coming very soon. You can now set cover and miniature separately.

**Reply by bc-devle** [reply]
why not fix in 8 ?
